### PR TITLE
Fix documentation for worker config

### PR DESCRIPTION
**Pull request by niketh**

### CODE REVIEW

File: docs/content/configuration/indexing-service.md [hunk 1]
```diff
@@ -173,8 +173,8 @@ Issuing a GET request at the same URL will return the current worker config spec
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`selectStrategy`|How to assign tasks to middle managers. Choices are `fillCapacity`, `fillCapacityWithAffinity`, `equalDistribution` and `javascript`.|fillCapacity|
-|`autoScaler`|Only used if autoscaling is enabled. See below.|null|
+|`worker.config.selectStrategy.type`|How to assign tasks to middle managers. Choices are `fillCapacity`, `fillCapacityWithAffinity`, `equalDistribution` and `javascript`.|fillCapacity|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: The old docs are wrong, but I think the new docs are wrong too. This block is meant to be documenting the JSON form of dynamic worker config, so the property-style "worker.config.selectStrategy.type" string shouldn't apply.

More correct for "selectStrategy" should be something like "this is an object, the options are documented below"

**Comment by niketh** [comment]
@gianm Changed the description

File: docs/content/configuration/indexing-service.md [hunk 2]
```diff
@@ -198,7 +198,7 @@ Workers are assigned tasks until capacity.
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`type`|`fillCapacity`.|required; must be `fillCapacity`|
+|`worker.config.selectStrategy.type`|`fillCapacity`.|required; must be `fillCapacity`|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: This should just be `type`, the original was correct

File: docs/content/configuration/indexing-service.md [hunk 3]
```diff
@@ -208,8 +208,8 @@ An affinity config can be provided.
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`type`|`fillCapacityWithAffinity`.|required; must be `fillCapacityWithAffinity`|
-|`affinity`|JSON object mapping a datasource String name to a list of indexing service middle manager host:port String values. Druid doesn't perform DNS resolution, so the 'host' value must match what is configured on the middle manager and what the middle manager announces itself as (examine the Overlord logs to see what your middle manager announces itself as).|{}|
+|`worker.config.selectStrategy.type`|`fillCapacityWithAffinity`.|required; must be `fillCapacityWithAffinity`|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: Original was correct here too

File: docs/content/configuration/indexing-service.md [hunk 4]
```diff
@@ -221,7 +221,7 @@ The workers with the least amount of tasks is assigned the task.
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`type`|`equalDistribution`.|required; must be `equalDistribution`|
+|`worker.config.selectStrategy.type`|`equalDistribution`.|required; must be `equalDistribution`|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: And here

File: docs/content/configuration/indexing-service.md [hunk 5]
```diff
@@ -234,8 +234,8 @@ its better to write a druid extension module with extending current worker selec
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`type`|`javascript`.|required; must be `javascript`|
-|`function`|String representing javascript function||
+|`worker.config.selectStrategy.type`|`javascript`.|required; must be `javascript`|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: And here

File: docs/content/configuration/indexing-service.md [hunk 6]
```diff
@@ -257,11 +257,11 @@ Amazon's EC2 is currently the only supported autoscaler.
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`minNumWorkers`|The minimum number of workers that can be in the cluster at any given time.|0|
-|`maxNumWorkers`|The maximum number of workers that can be in the cluster at any given time.|0|
-|`availabilityZone`|What availability zone to run in.|none|
-|`nodeData`|A JSON object that describes how to launch new nodes.|none; required|
-|`userData`|A JSON object that describes how to configure new nodes. If you have set druid.indexer.autoscale.workerVersion, this must have a versionReplacementString. Otherwise, a versionReplacementString is not necessary.|none; optional|
+|`worker.config.autoScaler.minNumWorkers`|The minimum number of workers that can be in the cluster at any given time.|0|
```
Comment (gianm) [inline docs/content/configuration/indexing-service.md]: And here

**Comment by gianm** [comment]
thx @niketh, posted another review.

**Comment by niketh** [comment]
@gianm Looking at the documentation its very hard to know that worker.config.selectStrategy.type should be used.to specify the type of selector strategy.
Also in the rest of the documentation the full type is mentioned, have the full property name mentioned there. For ex. druid.indexer.logs.s3Bucket or druid.indexer.logs.s3Bucket or druid.indexer.storage.type

I think we should keep worker.config.selectStrategy.type and not just type

**Pull request closed without merging**